lib/Parser/Parse.cpp

```diff
@@ -4609,6 +4609,12 @@ bool Parser::ParseFncDeclHelper(ParseNodePtr pnodeFnc, ParseNodePtr pnodeFncPare
         isEnclosedInParamScope = this->m_currentScope->GetEnclosingScope() && this->m_currentScope->GetEnclosingScope()->GetScopeType() == ScopeType_Parameter;
     }
 
+    if (isEnclosedInParamScope)
+    {
+        // We cannot just do HasDefaultArguments here becasue we may be inside a destructuring pattern.
+        Assert(pnodeFncParent && pnodeFncParent->sxFnc.HasNonSimpleParameterList());
+    }
```

Comment:
If the 'if' block is only for assert why not put isEnclosedInParamScope inside the assert itself?